symplyphysics/laws/thermodynamics/average_square_of_velocity.py

```diff
@@ -1,22 +1,39 @@
-from sympy import (Eq, solve)
+from sympy import (Eq, solve, integrate, S)
 from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
     validate_output, symbols, clone_symbol)
+from symplyphysics.core.expr_comparisons import expr_equals
+from symplyphysics.laws.thermodynamics.maxwell_boltzmann_distributions import speed_distribution
 
 # Description
-## For an ideal gas, the average square of velocity is directly proportional to its temperature and inversely proportional to the molar mass of the gas: <V^2> = 3 k T / m
-## Where:
+## For an ideal gas, the average square of velocity is directly proportional to its temperature
+## and inversely proportional to the molar mass of the gas.
+
+# Law: <V^2> = 3 k T / m
 ## <V^2> - average square of velocity
 ## T - temperature in gas
 ## m - mass of molecule
-## k - Boltzman's constant
-
-temperature_in_gas = Symbol("temperature_in_gas", units.temperature)
+## k - Boltzmann constant
 
-average_square_velocity = Symbol("average_square_velocity", units.velocity**2)
-molecule_mass = clone_symbol(symbols.basic.mass, "molecule_mass")
+average_square_velocity = Symbol("average_square_velocity", units.velocity**2, positive=True)
+temperature_in_gas = Symbol("temperature_in_gas", units.temperature, positive=True)
+molecule_mass = clone_symbol(symbols.basic.mass, "molecule_mass", positive=True)
 
 law = Eq(average_square_velocity, 3 * units.boltzmann_constant * temperature_in_gas / molecule_mass)
 
+# Derive from the Maxwell-Boltzmann speed distribution
+
+_distribution = speed_distribution.law.rhs.subs({
+    speed_distribution.particle_mass: molecule_mass,
+    speed_distribution.equilibrium_temperature: temperature_in_gas,
+})
+
+_average_square_of_speed_derived = integrate(
```

Comment:
Elaborate a bit on this formula. Why do you get average square velocity as a result?

Reply:
Will do